accord-core/src/test/java/accord/impl/basic/Cluster.java

```diff
@@ -496,7 +506,7 @@ private Function<List<Id>, BiFunction<Id, Id, Link>> overrideLinks(RandomSource
                 case RANDOM_BIDIRECTIONAL:
                 case RANDOM_UNIDIRECTIONAL:
                     boolean bidirectional = kind == RANDOM_BIDIRECTIONAL;
-                    int count = random.nextInt(bidirectional || random.nextBoolean() ? nodesList.size() : (nodesList.size() * nodesList.size())/2);
+                    int count = random.nextInt(bidirectional || random.nextBoolean() ? nodesList.size() : Math.max(1, (nodesList.size() * nodesList.size())/2));
```

Comment:
when nodesList size = 1, this fails (`1/2 = 0), hence the `Math.max(1` to keep it safe